flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionGraphHandler.java

```diff
@@ -81,23 +82,20 @@ public void reportCheckpointMetrics(
     public void reportInitializationMetrics(
             ExecutionAttemptID executionAttemptId,
             SubTaskInitializationMetrics initializationMetrics) {
-        if (executionGraph.getCheckpointStatsTracker() == null) {
+        final CheckpointCoordinatorConfiguration checkpointConfig =
+                executionGraph.getCheckpointCoordinatorConfiguration();
+        if (checkpointConfig == null || !checkpointConfig.isCheckpointingEnabled()) {
             // TODO: Consider to support reporting initialization stats without checkpointing
             log.debug(
                     "Ignoring reportInitializationMetrics if checkpoint coordinator is not present");
```

Comment:
Maybe tune the log message a bit. 
`Ignoring reportInitializationMetrics if checkpoint is disabled`

Reply:
I see where you're coming from but I am hesitant to do that for the following reasons:
* it's out-of-scope for this change
* I'm not 100% sure whether checkpoint disabled maps in all cases to the `CheckpointCoordinator` being not set. And I'm too lazy to do the code check for that one 😇

Reply:
Ok, the work on described in [my comment above](https://github.com/apache/flink/pull/24911#discussion_r1647484212) makes it necessary to verify the code. So, I might fix that along the way as well. :)